src/renderer/helpers/api/local.js

```diff
@@ -530,15 +538,20 @@ export function parseLocalTextRuns(runs, emojiSize = 16, options = { looseChanne
             const url = new URL(endpoint.payload.url)
             if (url.hostname === 'www.youtube.com' && url.pathname === '/redirect' && url.searchParams.has('q')) {
               // remove utm tracking parameters
-              const realURL = new URL(url.searchParams.get('q'))
+              const realURLStr = url.searchParams.get('q')
+              const realURL = new URL(realURLStr)
+              let anyUrlChanged = false
```

Comment:
Renaming this variable to `anyUrlChanges` or `urlChanged` might be better (`anyUrlChanged`, sounds like there are multiple URLs).